### DISCUSSION TITLE
Folder structure after compilation

### DISCUSSION THREAD

**Discussion by initiateit** [mamedev/discussions#24] [Q&A] [answered]
Sorry to ask such a silly question, but after getting through the hoops of compiling on Clear Linux I have the following folder structures:

![source](https://i.imgur.com/A0zpwRX.png)

and after compilation I have:

![compiled](https://i.imgur.com/qoeKpn5.png)

Is it correct to keep these folders only (along with the mame executable)?

<div class="w3-container w3-card-4 w3-padding-0 w3-border default-table" style="box-sizing: inherit; padding: 0px; box-shadow: rgba(0, 0, 0, 0.2) 0px 4px 10px 0px, rgba(0, 0, 0, 0.19) 0px 4px 20px 0px; border: 1px solid rgb(204, 204, 204) !important; width: 1776.5px; margin: 0px auto; color: rgb(0, 0, 0); font-family: Verdana, sans-serif; font-size: 15px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(253, 245, 230); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

Folder | Description
-- | --
artwork | Background artwork add the missing graphics used around the screen on arcade games.
bgfx | Contains the files for the bgfx video rendering system.
cfg | Contains xml based config files for systems that contain any changes you make on the in-game sliders.
ctrlr | This folder contains controller configuration files.
docs | Where the MAME readme files are stored.
hash | This is where MAME stores xml based software lists for certain drivers.
history | Created to hold an sql database by the MAME LUA scripting system.
hlsl | This is where hlsl effects and game specific settings are stored.
ini | Contains system initiation files based off of the main mame.ini.
inp | Where recorded game sessions are stored using the -record option.
language | This folder contains language files for the MAME interface.
nl-examples | Contains example files for MAME's net list functions.
nvram | This is where MAME saves EEPROM and RAM data from games
plugins | This folder contains scripts for the MAME LUA scripting system.
roms | Roms are the file MAME uses to load games. This is where they go.
samples | Some games have incomplete sound emulation so samples placed here are used instead
snap | This folder is where MAME can save screenshots and video files. Each system has it's own sub folder.
sta | Save state files are used to save your surrent position in a game. Each system has it's own sub folder.

</div><br class="Apple-interchange-newline">

Thanks for your help!

**Comment by cuavas** [comment, marked as answer]
Your list is pretty much correct. The only ones I might add are `keymaps` (can be used for friendly keyboard key names if you aren't using a US keyboard layout) and `web` (for the rarely-used HTTP server).

**Reply by initiateit** [reply]
Thanks kindly @cuavas , appreciated the help!
